rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from json import JSONDecodeError
+from typing import TYPE_CHECKING
+
+import requests
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import (
+    YEARN_VAULTS_V1_PROTOCOL,
+    YEARN_VAULTS_V2_PROTOCOL,
+    ChainID,
+    EvmTokenKind,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+YEARN_OLD_API = 'https://api.yearn.finance/v1/chains/1/vaults/all'
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_yearn_vaults(db: 'DBHandler') -> None:
+    """Query yearn API and ensure that all the tokens exists locally. If it exists but the protocol
+    is not the correct one the asset will be edited.
+    TODO: After Optimism gets merged we need to move this logic to a periodic task as we do with
+    curve pools.
+
+    May raise:
+    - RemoteError
+    """
+    try:
+        response = requests.get(YEARN_OLD_API)
+        data = response.json()
+    except requests.exceptions.RequestException as e:
+        raise RemoteError(f'Failed to obtain yearn vault information. {str(e)}') from e
+    except (DeserializationError, JSONDecodeError) as e:
+        raise RemoteError(f'Failed to deserialize data from cryptoscamdb. {str(e)}') from e
```

Comment:
:disappointed: copy pasta

```suggestion
        raise RemoteError(f'Failed to deserialize data from yearn's old api. {str(e)}') from e
```